com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/plugin/internal/ResolveMainClassHandler.java

```diff
@@ -0,0 +1,131 @@
+     */
+    public String[] resolveMainClass(List<Object> arguments) throws CoreException {
+        List<String> projectNames;
+        if (arguments.size() > 1 && arguments.get(1) != null) {
+            // project name specified
+            projectNames = new ArrayList<>();
+            projectNames.add((String) arguments.get(1));
+        } else {
+            List<String> projectFiles = (List<String>) arguments.get(0);
+            projectNames = resolveProjectName(projectFiles);
+        }
+        String[] res = {null, null};
+        for (String projectName : projectNames) {
+            String mainClass = resolveMainClassCore(projectName);
+            if (mainClass != null) {
+                res[0] = mainClass;
+                res[1] = projectName;
+                return res;
+            }
+        }
+        return res;
+    }
+
+    private List<String> resolveProjectName(List<String> projectFiles) {
+        List<String> projects = new ArrayList<>();
+        for (String f : projectFiles) {
+            SAXReader reader = new SAXReader();
+            Document document;
+            try {
+                document = reader.read(new File(f));
+                Element rootElm = document.getRootElement();
+                projects.add(rootElm.elementText("name"));
+            } catch (DocumentException e) {
+                logger.log(Level.WARNING, String.format("Exception on reading project file: %s.", f), e);
+            }
+        }
+        projects.add(null);
+        return projects;
+    }
+
+    private String resolveMainClassCore(String projectName) throws CoreException {
+        IJavaSearchScope searchScope = createSearchScope(projectName);
+        SearchPattern pattern = SearchPattern.createPattern("main(String[]) void", IJavaSearchConstants.METHOD,
+                IJavaSearchConstants.DECLARATIONS, SearchPattern.R_CASE_SENSITIVE | SearchPattern.R_EXACT_MATCH);
+        ArrayList<String> uris = new ArrayList<>();
+        SearchRequestor requestor = new SearchRequestor() {
+            @Override
+            public void acceptSearchMatch(SearchMatch match) {
+                Object element = match.getElement();
+                if (element instanceof IMethod) {
+                    IMethod method = (IMethod) element;
+                    try {
+                        if (method.isMainMethod()) {
+                            uris.add(method.getDeclaringType().getFullyQualifiedName());
+                        }
+                    } catch (JavaModelException e) {
+                        // ignore
+                    }
+                }
+            }
+        };
+        SearchEngine searchEngine = new SearchEngine();
+        searchEngine.search(pattern, new SearchParticipant[] {SearchEngine.getDefaultSearchParticipant()},
+                searchScope, requestor, null /* progress monitor */);
+        return uris.size() == 0 ? null : uris.get(0);
```

Comment:
if there are multiple main class in the target project, should pop up a dropdown menu in UI to tell user to choose one.

Reply:
I think we should return multiple to let js side has the possibilities to choose.